internal/services/cragcast/cragcast.go

```diff
@@ -0,0 +1,227 @@
+// this will contain functions for different forecast functionality.
```

Comment:
I'm a little surprised by the usage of `internal/` and the introduction of `services/cragcast`.

Could this simply live within `weather-client`? More-so than the logic (e.g. `GetForcast`), I think the types being in `internal` may be problematic over time?

Let me know your thoughts!

Reply:
After a deeper review, I do think we should consolidate this into `weather-client`. Additionally `internal/services/cragcast/types.go` should be moved back into `pkg/weather-client/types.go`.

Reply:
So the reason why i did this is because in my brain a client is something that fetches data from somewhere else. Basically our client(weather-client) will fetch the data from the service(noaa) and make it so our application doesn't need to do any fancy json to struct translating... it just gives us the data in a consumable format from the api.

In my mind the translation of the raw data to a format that our application likes should happen in our application, not as part of a client.

I do get the merits of having the client do the translation so that whatever service we're using(noaa, wunderground, weather.com) will feed our application the formatted data it wants so the application layer doesnt need a bunch of different translators for different services.

Your way just makes the clients a little bit more complex to implement whereas my version makes the client's job simpler.

I don't have strong opinions given the tradeoffs I could go either way. Your thoughts?